local_adaptivity/fed_avg_local_adaptivity.py

```diff
@@ -208,32 +220,36 @@ def client_update(model,
 
     model_weights = _get_weights(model)
     tff.utils.assign(model_weights, initial_weights)
-
     num_examples = tf.constant(0, dtype=tf.int32)
-    # Need to replace names of following two variables.
-    m_states = tf.nest.map_structure(tf.zeros_like, model_weights.trainable)
-    n_states = tf.nest.map_structure(tf.zeros_like, model_weights.trainable)
+
+    # Initialize local states for local and global corrections
+    avg_local_states = tf.nest.map_structure(tf.zeros_like, model_weights.trainable)
+    cum_local_states = tf.nest.map_structure(tf.zeros_like, model_weights.trainable)
     for batch in iter(dataset):
       with tf.GradientTape() as tape:
         output = model.forward_pass(batch)
       grads = tape.gradient(output.loss, model_weights.trainable)
       grads_and_vars = zip(grads, model_weights.trainable)
       client_optimizer.apply_gradients(grads_and_vars)
       num_examples += tf.shape(output.predictions)[0]
+
+      # Get momentum factor and preconditioner to update local states
       client_opt_beta = _get_optimizer_momentum_beta(client_optimizer)
       client_opt_preconditioner = _get_optimizer_preconditioner(
           client_optimizer, model_weights)
-
-      m_states = tf.nest.map_structure(
+      avg_local_states = tf.nest.map_structure(
           lambda m, p, b=client_opt_beta: b * m + (1 - b) * p,
-          m_states,
+          avg_local_states,
           client_opt_preconditioner)
-      n_states = tf.nest.map_structure(lambda m, n: m + n, m_states, n_states)
+      cum_local_states = tf.nest.map_structure(
+          lambda m, n: m + n, avg_local_states, cum_local_states)
 
     aggregated_outputs = model.report_local_outputs()
     weights_delta = tf.nest.map_structure(
         lambda a, b, c: tf.math.divide_no_nan(a - b, c),
-        model_weights.trainable, initial_weights.trainable, n_states)
+        model_weights.trainable, initial_weights.trainable, cum_local_states)
+    cum_local_states = tf.nest.map_structure(
```

Comment:
It is a bit confusing to use `cum_local_states` to represent both "preconditioner" and the "inverse of preconditioner".